pyroma/testdata/complete/setup.py

```diff
@@ -10,6 +10,7 @@
     version=version,
     description="This is a test package for pyroma.",
     long_description=long_description,
+    python_requires=">=2.6",
```

Comment:
For completeness, why not have a more complex (and realistic) specifierset here (and the others)?

```suggestion
    python_requires=">=2.6,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*",
```

Reply:
> I don't really understand the motivation for gloming this functionality on to the existing classifier test when this is really an entirely separate, independent check. 

I initially started it as an independent test like in your snippet. However, at least as I understood it reading the code, pyroma used `classfiers ` for python version checking as the **reference**, ie if you had a "python version classifier" then your lib was "well defined" as a python package.

If it was up to me, I would have moved the _classifier python version check_ to _advisory only_ (ie the test would not contribute to the '10 points' rating) and used only the _python-requires check_ for _rating_ purpose because that's what pip, poetry and others tools use.

My opinion is irrelevant though: pyroma users have been using classifiers this way. 

**I do not wish to break package rating** without first informing users. That's the maintainer's job to decide how / when to do this + communicate this to end users (deprecation policy, etc).

This decision has not been made. As such, I did not see how to both checks independently without breaking existing behaviour.

> This adds technical risk to this PR, making reviewing and testing it more difficult and increases future maintenance burden, for little clear gain.

The existing code has been improved (clearer return check / weight ).

> just complicates the logic and implementation of the existing check and furthermore makes what is already the longest and most complicated check even longer and more complicated

Well, opinion differs here. Hopefully, the first 3 paragraph clarify my intent.

**Willing to discuss this further and revert to your preferred solution.**

Reply:
> For completeness, why not have a more complex (and realistic) specifierset here (and the others)?

Testing a simple case is sufficient to illustrate pyroma perform the right check. More complex case would not run additional pyroma lines of code (but will in packaging.specifiers which is tasked to verify the version requirements).

Reply:
> MO, I confess that I don't really understand the motivation for gloming this functionality on to the existing classifier test when this is really an entirely separate, independent check. Conflated them, as well as implementing rather complex handling of various interactions between them, just complicates the logic and implementation of the existing check and furthermore makes what is already the longest and most complicated check even longer and more complicated. This adds technical risk to this PR, making reviewing and testing it more difficult and increases future maintenance burden, for little clear gain.

If I do what you suggested, then most test fails with 'You should specify what Python version ... with 'python_requires=...''. 

Not sure you guys want to break user packages in the next release (even though it is the right thing to do).

Reply:
Side note: we are not the only ones to bump into [the classifier for humans , python_requires for machine](https://github.com/badges/shields/issues/5551).

Reply:
> > For completeness, why not have a more complex (and realistic) specifierset here (and the others)?
> 
> Testing a simple case is sufficient to illustrate pyroma perform the right check. More complex yet valid case would not run additional pyroma lines of code (but will in `packaging` package which is tasked to verify various version requirements syntax).

Yet, I still added case for good measure ...